components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/dao/ParMgtDAOImpl.java

```diff
@@ -0,0 +1,136 @@
+import org.wso2.carbon.identity.oauth.par.exceptions.ParCoreException;
+import org.wso2.carbon.identity.oauth.par.model.ParRequestDO;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Map;
+
+/**
+ * Implementation of abstract DAO layer.
+ */
+public class ParMgtDAOImpl implements ParMgtDAO {
+
+    private static final Log log = LogFactory.getLog(ParMgtDAOImpl.class);
+    ObjectMapper objectMapper = new ObjectMapper();
+
+    @Override
+    public void persistParRequest(String reqUriUUID, String clientId, long scheduledExpiryTime,
+                                  Map<String, String> parameters) throws ParCoreException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+
+            try (PreparedStatement prepStmt = connection.prepareStatement(SQLQueries.
+                    ParSQLQueries.STORE_PAR_REQUEST)) {
+
+                prepStmt.setString(1, reqUriUUID);
+                prepStmt.setString(2, clientId);
+                prepStmt.setLong(3, scheduledExpiryTime);
+                prepStmt.setString(4, getJsonParams(parameters));
+
+                prepStmt.execute();
+            } catch (SQLException e) {
+                IdentityDatabaseUtil.rollbackTransaction(connection);
+                throw new ParCoreException("Error occurred in persisting the successful PAR request with" +
+                        " uuid: " + reqUriUUID);
+            }
+        } catch (SQLException e) {
+            throw new ParCoreException("Error occurred in persisting the successful PAR request with" +
+                    " uuid: " + reqUriUUID);
+        }
+    }
+
+    @Override
+    public ParRequestDO getParRequest(String uuid) throws ParCoreException {
+
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+             PreparedStatement prepStmt = connection.prepareStatement(SQLQueries
+                     .ParSQLQueries.RETRIEVE_PAR_REQUEST)) {
+
+            prepStmt.setString(1, uuid);
+
+            try (ResultSet resultSet = prepStmt.executeQuery()) {
+                if (resultSet.next()) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Successfully obtained client_id of RequestURI with UUID: " + uuid);
+                    }
+                    String jsonParams = resultSet.getString("JSON_PARAMS");
+                    long scheduledExpiry = resultSet.getLong("SCHEDULED_EXPIRY");
+                    String clientId = resultSet.getString("CLIENT_ID");
+
+                    return new ParRequestDO(getParams(jsonParams), scheduledExpiry, clientId);
+
+                } else {
+                    if (log.isDebugEnabled()) {
+                        log.debug("PAR request with UUID " + uuid + " does not exist");
+                    }
+                    throw new ParCoreException("uuid does not exist");
```

Comment:
Since we are throwing an exception we dont need a debug log here ? and better make error description explanatory